shim.go

```diff
@@ -96,6 +98,36 @@ func newShimConfig(config PodConfig) interface{} {
 	}
 }
 
+func shimRunning(pid int) (bool, error) {
+	process, err := os.FindProcess(pid)
+	if err != nil {
+		return false, err
+	}
+
+	if err := process.Signal(syscall.Signal(0)); err != nil {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func stopShim(pid int) error {
+	running, err := shimRunning(pid)
```

Comment:
This is racy. Go doesn't exactly make this easy, but I think what you need to do is:

- Call `syscall.RawSyscall` directly.
- Check the returned value (which might be `syscall.ESRCH`).